src/ClusterBootstrap/ctl.py

```diff
@@ -253,6 +254,67 @@ def render_template_or_dir(config, args):
         utils.render_template(src, dst, config)
 
 
+def maintain_db(config, args):
+    """
+    push/pull a table to/from DB
+    """
+    subcommand = args.nargs[0]
+    assert subcommand in ["pull", "push", "connect"], "invalid action."
+    host = config["mysql_node"]
+    user = config["mysql_username"]
+    password = config["mysql_password"]
+    if subcommand == "connect":
+        os.system("mysql -h {} -u {} -p{}".format(host, user, password))
+    else:
+        database = "DLWSCluster-{}".format(config["clusterId"])
+        table_name = args.nargs[1]
+        assert table_name in ["vc", "acl"], "invalid table."
+        
+        if args.verbose:
+            print("connecting to {}@{}, DB {}".format(user, host, database))
+        conn = connector.connect(user=user, password=password,
+               host=host, database=database)
+        if subcommand == "pull":
+            sql = "SELECT * from {}".format(table_name)
+            cursor = conn.cursor()
+            cursor.execute(sql)
+            col_names = [col[0] for col in cursor.description]
+            serialized_rows = []
+            rows = cursor.fetchall()
+            for row in rows:
+                serialized_row = {}
+                for i, v in enumerate(row):
+                    try:
+                        serialized_row[col_names[i]] = json.loads(v)
+                    # JSONDecodeError
+                    except:
+                        serialized_row[col_names[i]] = v
+                serialized_rows.append(serialized_row)
+            table_config = {"col_names": col_names, "rows": serialized_rows}
+            dump2 = args.output if args.output else "{}.yaml".format(table_name)
+            with open(dump2, "w") as wf:
+                yaml.safe_dump(table_config, wf)
+        elif subcommand == "push":
+            sql = "DELETE from {}".format(table_name)
+            cursor = conn.cursor()
+            cursor.execute(sql)
+            with open("{}.yaml".format(table_name)) as f:
```

Comment:
Symmetrically, I think we should handle user specified `args.in`.

Reply:
updated